jupyterlab_bigquery/src/components/details_panel/details_panel.tsx

```diff
@@ -80,18 +83,63 @@ interface SharedDetails {
   labels: string[];
   name: string;
   schema?: SchemaField[];
+  query?: string;
 }
 
 interface Props {
   details: SharedDetails;
   rows: any[];
   // TODO(cxjia): figure out a shared typing for these rows
-  detailsType: string;
+  detailsType: 'DATASET' | 'TABLE' | 'VIEW';
 }
 
+const getTitle = type => {
+  switch (type) {
+    case 'DATASET':
+      return 'Dataset info';
+    case 'TABLE':
+      return 'Table info';
+    case 'VIEW':
+      return 'View info';
+  }
+};
+
 export const DetailsPanel: React.SFC<Props> = props => {
   const { details, rows, detailsType } = props;
 
+  function handleEditorDidMount(_, editor) {
+    const editorElement = editor.getDomNode();
+    if (!editorElement) {
+      return;
+    }
+
+    monaco
+      .init()
+      .then(monaco => {
+        const lineHeight = editor.getOption(
+          monaco.editor.EditorOption.lineHeight
+        );
+        const lineCount = editor._modelData.viewModel.getLineCount();
+        const height = lineCount * lineHeight;
+        editorElement.style.height = `${height}px`;
+        editor.layout();
```

Comment:
This gets the container to be the correct height, but for some reason there is extra contents beyond the line numbers and actual query text. I will continue investigating, but for now all the relevant query text shows up on the screen ok.

Reply:
Aha! Resolved with the editor option `scrollBeyondLastLine: false`